From 9f6e148737dd1e12f37fe9fd769e50869b643e1e Mon Sep 17 00:00:00 2001
From: Francesco Pham <francesco.pham1@h-partners.com>
Date: Thu, 28 May 2026 11:59:04 +0200
Subject: [PATCH 2/2] storage_service: disable TEE client and HUKS HDI on
 x86_general

x86_general has no secure-element / TEE hardware; mirror the existing is_emulator gate so the build does not try to link against TEE/HUKS HDIs that have no implementation.

Change-Id: Ic21f7bd3704ad0a82f5c16a88427a86e9bb9172c
Signed-off-by: Francesco Pham <francesco.pham1@h-partners.com>
---
 storage_service_aafwk.gni | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/storage_service_aafwk.gni b/storage_service_aafwk.gni
index 30b7f49e5..a7ce28ad6 100644
--- a/storage_service_aafwk.gni
+++ b/storage_service_aafwk.gni
@@ -41,8 +41,8 @@ declare_args() {
                                defined(global_parts_info.thirdparty_libfuse)
   storage_service_dfs_filemanager = defined(global_parts_info) &&
                                     defined(global_parts_info.filemanagement_dfs_service)
-  enable_tee_client = !is_emulator
-  enable_huks_hdi = !is_emulator
+  enable_tee_client = !is_emulator && product_name != "x86_general"
+  enable_huks_hdi = !is_emulator && product_name != "x86_general"
 }
 
 storage_manager_path =
-- 
2.50.1

